File: src/main/kotlin/acc/integration/AccountDAO.java
package acc.integration;

import acc.integration.impl.AcountDAODefault;
import acc.model.AnalAcc;
import acc.model.AccGroup;
import acc.model.AccId;
import acc.util.AccException;
import java.util.List;
import java.util.Optional;

public abstract class AccountDAO {

    public static AccountDAO instance = new AcountDAODefault();

    public abstract void create(AccGroup skupina, String no, String name) throws AccException;

    public abstract List<AnalAcc> getAll() throws AccException;

    public abstract Optional<AnalAcc> getByNumber(String name) throws AccException;

    public abstract AnalAcc getPocatecniUcetRozvazny() throws AccException;

    public abstract void delete(AccId id) throws AccException;

    public abstract List<AnalAcc> getBalancing() throws AccException;

    public abstract List<AnalAcc> getByGroup(AccGroup accg) throws AccException;

    public abstract List<AnalAcc> getByClass(AccGroup accg) throws AccException;

}

File: src/main/kotlin/acc/integration/AccountDAO.kt
package acc.integration

import acc.integration.impl.AcountDAODefault
import acc.model.AnalAcc
import acc.model.AccGroup
import acc.model.AnalId
import acc.util.AccException
import java.util.Optional

abstract class AccountDAO {

    abstract val all: List<AnalAcc>

    abstract val pocatecniUcetRozvazny: AnalAcc

    abstract val balancing: List<AnalAcc>

    @Throws(AccException::class)
    abstract fun create(skupina: AccGroup, no: String, name: String)

    @Throws(AccException::class)
    abstract fun getByNumber(name: String): Optional<AnalAcc>

    @Throws(AccException::class)
    abstract fun delete(id: AnalId)

    @Throws(AccException::class)
    abstract fun getByGroup(accg: AccGroup): List<AnalAcc>

    @Throws(AccException::class)
    abstract fun getByClass(accg: AccGroup): List<AnalAcc>

    companion object {

      //  var instance: AccountDAO = AcountDAODefault()
    }

}
